R/packages.R

```diff
@@ -0,0 +1,26 @@
+# Dependency: must be working with an RStudio Project
+# TODO: Remove packrat dependency; initial run of packrat::init is overkill
+
+library(packrat)
+library(dplyr)
+
+source("utils.R")
+
+get_packages_list <- function(project_dir = getwd()) {
+  # Init or snapshot to create or update the packrat.lock file
+  lock_file <- paste0(project_dir, "/packrat/packrat.lock")
+  if (!file.exists(lock_file)) {
+    packrat::init(project = project_dir, options = list(), enter = FALSE)
+  } else {
+    packrat::snapshot(project_dir, prompt = FALSE, snapshot.sources = FALSE, infer.dependencies = TRUE)
```

Comment:
I think we could use `dry.run` in `packrat::snapshot` to avoid writing a `packrat.lock` file and assign the output to an object.

Reply:
That should work for `snapshot` but not `init`; I'll test and update.